dp/cloud/python/magma/radio_controller/tests/unit/test_active_mode_controller_service.py

```diff
@@ -46,169 +48,360 @@ def setUp(self):
+            with_eirp_capabilities(0, 10, 20, 1).\
+            with_desired_state(Registered).\
+            build()
+        other_config = ActiveModeConfigBuilder(). \
+            with_state(Registered). \
+            with_registration('other'). \
+            with_eirp_capabilities(5, 15, 25, 3). \
+            with_desired_state(Registered). \
+            build()
+        expected = State(active_mode_configs=[some_config, other_config])
+
+        actual = self.amc_service.GetState(GetStateRequest(), None)
+        self.assertEqual(expected, actual)
+
+    def test_not_get_state_without_active_mode_config(self):
+        cbsd = DBCbsdBuilder(). \
+            with_state(self.unregistered). \
+            with_registration('some'). \
+            with_eirp_capabilities(0, 10, 20, 1). \
+            build()
+        self.session.add(cbsd)
+        self.session.commit()
+
+        expected = State()
+
+        actual = self.amc_service.GetState(GetStateRequest(), None)
+        self.assertEqual(expected, actual)
 
-        some_cbsd = DBCbsd(
-            id=1,
-            state_id=cbsd_states[CbsdStates.REGISTERED.value],
-            cbsd_id="some_cbsd_id",
-            user_id="some_user_id",
-            fcc_id="some_fcc_id",
-            cbsd_serial_number="some_serial_number",
-            eirp_capability=26.5,
-            last_seen=datetime.fromtimestamp(1234),
+    def test_not_get_state_without_registration_params(self):
+        cbsd = DBCbsdBuilder(). \
+            with_state(self.unregistered). \
+            with_eirp_capabilities(0, 10, 20, 1). \
+            with_active_mode_config(self.registered).\
+            build()
+        self.session.add(cbsd)
+        self.session.commit()
+
+        expected = State()
+
+        actual = self.amc_service.GetState(GetStateRequest(), None)
+        self.assertEqual(expected, actual)
+
+    def test_not_get_state_without_eirp_capabilities(self):
+        cbsd = DBCbsdBuilder(). \
+            with_state(self.unregistered). \
+            with_registration('some'). \
+            with_active_mode_config(self.registered). \
+            build()
+        self.session.add(cbsd)
+        self.session.commit()
+
+        expected = State()
+
+        actual = self.amc_service.GetState(GetStateRequest(), None)
+        self.assertEqual(expected, actual)
+
+    def _prepare_base_cbsd(self) -> DBCbsdBuilder:
+        return DBCbsdBuilder(). \
+            with_state(self.unregistered). \
+            with_registration('some'). \
+            with_eirp_capabilities(0, 10, 20, 1). \
+            with_active_mode_config(self.registered)
+
+    @staticmethod
+    def _prepare_base_active_mode_config() -> ActiveModeConfigBuilder:
+        return ActiveModeConfigBuilder(). \
+            with_state(Unregistered). \
+            with_registration('some'). \
+            with_eirp_capabilities(0, 10, 20, 1). \
+            with_desired_state(Registered)
+
+
+class ActiveModeConfigBuilder:
```

Comment:
pretty neat :) maybe it'd be a good idea to take these builders out to a new module - test_utils or sth

Reply:
moved to test_utils